File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        st = []
        count = 0
        for c in s:
            if c is '(':
                st.append(c)
            else:
                if not st:
                    count += 1
                else:
                    st.pop()
        return len(st) + count

File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        Stack<Character> st = new Stack<>();
        int count = 0;
        for (char c : s.toCharArray()) {
            if (c == '(') {
                st.push(c);
            } else {
                if (st.isEmpty()) {
                    count++;
                } else {
                    st.pop();
                }
            }
        }
        return st.size() + count;
    }
}